data-pipeline/src/trace_exporter/mod.rs

```diff
@@ -640,6 +647,9 @@ impl TraceExporter {
                 let send_data = SendData::new(size, tracer_payload, header_tags, &endpoint);
                 self.runtime.block_on(async {
                     let send_data_result = send_data.send().await;
+                    if let Some(telemetry) = &self.telemetry {
+                        telemetry.send(&send_data_result).await;
+                    }
```

Comment:
Are you sure we want to do this synchronously with the trace export?
I think that in term of qos, traces should not be lagging if telemetry metrics are, for any reasons and I'd rather see us use the non async, non blocking methods on the TelemetryWorkerHandle (which might drop points if there is backpressure)

Reply:
+1, I thought we discussed that we'd be sending telemetry async, unless I misunderstood something.

Reply:
Just to give some context to the discussion, the 'send' method just adds the points to the corresponding bucket inside the telemetry client. I think this is done through a tokio channel. The sending itself it's done in a separate tokio task which is spawned at the trace exporter instantiation and is triggered at a certain interval which is configurable through the trace exporter configuration. Having said that, Paul are you referring to the fact that we can overwhelm the tokio, running out of its capacity and trigger some waiting on the current task?

Reply:
> Paul are you referring to the fact that we can overwhelm the tokio, running out of its capacity and trigger some waiting on the current task?

Yes, if I remember correctly (and it hasn't changed), the telemetry sender is structured as a dispatch loop pulling data (like points) and events (like flush metrics) from the channel.
There is a lot of buffering for the channel (5K messages) but of the http requests are lagging a lot it might still be full and block traces.

What I'm sugesting is that instead of using the async methods on the TelemetySender, which wait for space to be available in the channel, we use the sync ones (like `TelemetryWorkerHandle::add_point`, which use try_send and will drop points  if the channel is full)